online-shopping-website/src/Components/ProductDetails.js

```diff
@@ -0,0 +1,128 @@
+        this.setState({quantity: e.target.value});
+    }
+
+    get quantity() {
+        return this.state.quantity;
+    }
+
+    render() {
+        return (
+            <div className="ProductDetails-QuantityButtonsContainer">
+                <h3>Quantity</h3>
+                <Stack className="ProductDetails-QuantityButtonsStack" direction="row" spacing={1}>
+                    <Button className="ProductDetails-QuantityButtons" variant="contained"
+                            disabled={this.state.quantity == 10}
+                            onClick={this.IncrementItem}>
+                        <AddIcon/>
+                    </Button>
+                    <input className="inputne" value={this.state.quantity} onChange={this.UpdateValue}/>
+                    <Button className="ProductDetails-QuantityButtons" variant="contained"
+                            disabled={this.state.quantity == 1}
+                            onClick={this.DecreaseItem}>
+                        <RemoveIcon/>
+                    </Button>
+                </Stack>
+                <h5 className="ProductDetails-ProductLimitText">Limit of 10 items per product in cart.</h5>
+                <Button className="ProductDetails-CartButton" variant="contained" endIcon={<AddShoppingCartIcon/>}>
+                    Add to cart
+                </Button>
+            </div>
+        );
+    }
+}
+
+export const ProductDetails = () => {
+    const productName = useParams();
+
+    const selectedProduct = {
+        brand: "",
+        description: "",
+        id: null,
+        image: "",
+        name: "",
+        price: null,
+        seller: ""
+    }
+
+    Products.products.map(product => {
+        if (productName.productName === product.name) {
+            selectedProduct.brand = product.brand;
+            selectedProduct.description = product.description;
+            selectedProduct.id = product.id;
+            selectedProduct.image = product.image;
+            selectedProduct.name = product.name;
+            selectedProduct.price = product.price.toFixed(2);
+            selectedProduct.seller = product.seller;
+        }
+    });
+
+    return (
+        <Grid container className="ProductDetails-Container">
+            <Grid item xs={12} sm={9} lg={8}>
+                <h1>{selectedProduct.name}</h1>
+                <img className="ProductDetails-Image" src={selectedProduct.image} alt={selectedProduct.name}/>
+
+                <Grid item container lg={12}>
+                    <Grid item conatiner lg={6}>
+                            <h3>Brand</h3>
+                            <h4>{selectedProduct.brand}</h4>
+                    </Grid>
+
+                    <Grid item lg={6}>
+                        <h3>Seller</h3>
+                        <h4>{selectedProduct.seller}</h4>
+                    </Grid>
+                </Grid>
+
+
+                <h3>Description</h3>
+                <h4>{selectedProduct.description}</h4>
```

Comment:
The description should probably be a `<p>` instead of a heading. For some products, the description might be pretty long so that shouldn't be emphasized.